File: src/spawners/BasicSpawner.cpp
#include "BasicSpawner.hpp"
#include "Environment.hpp"
#include "BasicEnemy.hpp"
#include <iostream>

int BasicSpawner::FIRE_TIME = 60;

void BasicSpawner::render(sf::RenderWindow& window) {
	sf::RectangleShape shape(sf::Vector2f(16, 20));
	shape.setPosition(m_position);
	shape.setFillColor(sf::Color(35, 38, 36));
	window.draw(shape);
}

void BasicSpawner::spawnEnemy() {
	Enemy* enemy = new BasicEnemy(m_position.x + 3, m_position.y + 20);
	m_env->addEnemy(enemy);
}

File: bin/Debug/mods/GameData/spawners/BasicSpawner.lua
BasicSpawner = {
	fireTime = 60,
	fill = Color.new(35, 38, 36)
}

function BasicSpawner.spawnEnemies(spawner)
	return {
		amount = 1,
		enemies = {
			Enemy.new(spawner.position.x + 3, spawner.position.y + 20, 0.0, 2.0, 300, Color.new(35, 38, 36))
		}
	}
end